adapter/src/ethereum.rs

```diff
@@ -269,6 +271,64 @@ impl Adapter for EthereumAdapter {
         ewt_sign(&wallet, &self.keystore_pwd, &payload)
             .map_err(|err| AdapterError::Adapter(Error::SignMessage(err).into()))
     }
+
+
+    async fn get_spendable(&self, channel: &Channel, spender: &ValidatorId) -> AdapterResult<SpendableOutput, Self::AdapterError> {
+        let outpace_contract = Contract::from_json(
+            self.web3.eth(),
+            self.config.ethereum_core_address.into(),
+            &OUTPACE_ABI,
+        )
+        .map_err(Error::ContractInitialization)?;
+
+        let erc20_contract = Contract::from_json(
+            self.web3.eth(),
+            self.config.ethereum_core_address.into(),
+            &ERC20_ABI,
+        )
+        .map_err(Error::ContractInitialization)?;
+
+        let mut amount: U256 = tokio_compat_02::FutureExt::compat(async {
+            tokio_compat_02::FutureExt::compat(outpace_contract.query(
+                "deposits",
+                (channel.id.into_token(), spender.to_hex_prefix_string().into_token()),
+                None,
+                Options::default(),
+                None,
+            ))
+            .await
+        })
+        .await
+        .map_err(Error::ContractQuerying)?;
+
+        let to_be_deposited: U256 = tokio_compat_02::FutureExt::compat(async {
+            tokio_compat_02::FutureExt::compat(erc20_contract.query(
+                "balanceOf",
+                channel.deposit_asset.clone().into_token(),
```

Comment:
This should probably also change, because we use `Address` for the token address in the new `Channel`.
The types right now are different, the contract expects an `Address`, however we pass it a `Token::String`